tools/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/ApplicationInfo.scala

```diff
@@ -34,16 +34,165 @@ import org.apache.spark.deploy.history.{EventLogFileReader, EventLogFileWriter}
 import org.apache.spark.internal.Logging
 import org.apache.spark.io.CompressionCodec
 import org.apache.spark.scheduler._
-import org.apache.spark.sql.{DataFrame, SparkSession}
+import org.apache.spark.sql.{DataFrame, Row, SparkSession}
 import org.apache.spark.sql.execution.SparkPlanInfo
+import org.apache.spark.sql.execution.metric.SQLMetricInfo
 import org.apache.spark.sql.execution.ui.SparkPlanGraph
+import org.apache.spark.sql.types.StructType
 import org.apache.spark.ui.UIUtils
 import org.apache.spark.util._
 
+class SparkPlanInfoWithStage(
+    nodeName: String,
+    simpleString: String,
+    override val children: Seq[SparkPlanInfoWithStage],
+    metadata: scala.Predef.Map[String, String],
+    metrics: Seq[SQLMetricInfo],
+    val stageId: Option[Int]) extends SparkPlanInfo(nodeName, simpleString, children,
+  metadata, metrics) {
+
+  import SparkPlanInfoWithStage._
+
+  def debugEquals(other: Any, depth: Int = 0): Boolean = {
```

Comment:
I left this in, because our plugin is constantly changing and we have not really tested this with databricks or others, so if we run into issues, this is a quick way to track down what parts of a plan need to be normalized for the comparison to work properly.